common/math/osqp_interface/CMakeLists.txt

```diff
@@ -1,50 +1,39 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(osqp_interface)
 
-add_compile_options(-std=c++14 -Ofast)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
-  rostest
-  rosunit
-)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Ofast)
+endif()
+
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
 find_package(Eigen3 REQUIRED)
 
+find_package(osqp_vendor REQUIRED)
+
 find_package(osqp REQUIRED)
-get_target_property(osqp_INCLUDE_DIR osqp::osqpstatic INTERFACE_INCLUDE_DIRECTORIES)
-
-catkin_package(
-  INCLUDE_DIRS
-    include
-    ${osqp_INCLUDE_DIR}
-  LIBRARIES
-    osqp_interface
-)
-
-include_directories(
-  include
-  ${EIGEN3_INCLUDE_DIR}
-  ${osqp_INCLUDE_DIR}
-  ${catkin_INCLUDE_DIRS}
-)
-
-add_library(osqp_interface src/osqp_interface.cpp src/csc_matrix_conv.cpp)
-
-target_link_libraries(osqp_interface
-  osqp::osqpstatic
-  ${catkin_LIBRARIES}
-)
-
-install(
-  TARGETS
-    osqp_interface
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
-
-## Install project namespaced headers
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-)
+get_target_property(OSQP_INCLUDE_DIR osqp::osqpstatic INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
I've tested locally, and I could remove this and just use `target_link_libraries(osqp_interface osqp::osqpstatic)`.

Reply:
Yeah but that won't work if you use the `osqp_interface` from say `mpc_follower` because the interface publicly `#include <osqp.h>`, so we need to get the transitive includes exported as well

Reply:
Good point, makes sense.